app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ApplicationsController < ApplicationController
+  def new
+  end
+
+  def create
+    application = Application.new(application_params)
+    if application.save
+      redirect_to "/applications/#{application.id}"
+    else
+      flash.now[:notice] = "Error: #{error_message(application.errors)}"
+      render :new
+    end
+  end
+
+  def show
+    @application = Application.find(params[:id])
+    if params[:search].present?
+      @pets = Pet.search(params[:search])
+    end
+  end
+
+  def update
+    application = Application.find(params[:id])
+    application.update(application_params)
+    application.update(status: "Pending")
+    redirect_to "/applications/#{application.id}"
+  end
```

Comment:
Instead of having two separate update operations which would both make separates calls to the database, you can reduce this to one call by first adding to the `application_params` and then updating in one call.

```ruby
def update
  application = Application.find(params[:id])
  application.update(application_params.merge({ status: "Pending" }))
  redirect_to "/applications/#{application.id}"
end
```

This would also work:

```ruby
def update
  application = Application.find(params[:id])
  application_params[:status] = "Pending"
  application.update(application_params)
  redirect_to "/applications/#{application.id}"
end
```

We're just trying to limit the number of times we hit the database.